integration/common.go

```diff
@@ -0,0 +1,112 @@
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/pelletier/go-toml"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cri "k8s.io/cri-api/pkg/apis"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// RegistryList holds public image registries
+type RegistryList struct {
+	DockerLibraryRegistry    string
+	K8sCriContainerdRegistry string
+	K8sGcrRegistry           string
+}
+
+var (
+	imageService             cri.ImageManagerService
+	imageConfigs             map[int]string
+	dockerLibraryRegistry    string
+	k8sCriContainerdRegistry string
+	k8sGcrRegistry           string
+	pauseImage               string // This is the same with default sandbox image
+)
+
+func initReg(repoList string) {
+	registry := RegistryList{
+		DockerLibraryRegistry:    "docker.io/library",
+		K8sCriContainerdRegistry: "gcr.io/k8s-cri-containerd",
+		K8sGcrRegistry:           "k8s.gcr.io",
+	}
+
+	if repoList != "" {
+		fileContent, err := ioutil.ReadFile(repoList)
+		if err != nil {
+			panic(fmt.Errorf("Error reading '%v' file contents: %v", repoList, err))
+		}
+
+		err = toml.Unmarshal(fileContent, &registry)
+		if err != nil {
+			panic(fmt.Errorf("Error unmarshalling '%v' TOML file: %v", repoList, err))
+		}
+	}
+
+	dockerLibraryRegistry = registry.DockerLibraryRegistry
+	k8sCriContainerdRegistry = registry.K8sCriContainerdRegistry
+	k8sGcrRegistry = registry.K8sGcrRegistry
+
+	imageConfigs = initImageConfigs()
+	pauseImage = GetImage(Pause)
+}
+
+const (
+	// None is to be used for unset/default images
+	None = iota
+	// BusyBox image
+	BusyBox
+	// Pause image
+	Pause
+	// VolumeCopyUp image
+	VolumeCopyUp
+	// VolumeOwnership image
+	VolumeOwnership
+)
+
+func initImageConfigs() map[int]string {
+	configs := map[int]string{}
+	configs[BusyBox] = fmt.Sprintf("%s/busybox:1.29", dockerLibraryRegistry)
+	configs[Pause] = fmt.Sprintf("%s/pause:3.5", k8sGcrRegistry)
+	configs[VolumeCopyUp] = fmt.Sprintf("%s/volume-copy-up:1.0", k8sCriContainerdRegistry)
+	configs[VolumeOwnership] = fmt.Sprintf("%s/volume-ownership:1.0", k8sCriContainerdRegistry)
+	return configs
+}
+
+// GetImage returns the fully qualified URI to an image (including version)
+func GetImage(image int) string {
+	return imageConfigs[image]
+}
+
+// PullImageWithCleanup pulls the given image into the given sandbox, ensures that no error was encountered
+// while pulling it, and sets up a cleanup to remove it after the test ends.
+func PullImageWithCleanup(t *testing.T, imageName string, config *runtime.PodSandboxConfig) {
```

Comment:
I think it's fine to keep this test where it was since this is never used except where it was?

Reply:
Sure, other PRs are going to need it as well, but I suppose it can be added in them instead. Removed.